103_agent-driven-attestation.md

```diff
@@ -0,0 +1,421 @@
+
+> The TPM reports on the state of the platform by quoting the PCR values. For assurance that these PCR values accurately reflect that state, it is necessary to establish the binding between the RTR and the platform.
+
+As such, we have two different trust concerns:
+
+- **The need to identify a TPM as having been produced by a trusted manufacturer.** In order to have confidence in measurements of system state, it is necessary that the TPM behaves according to spec, e.g., that it correctly performs PCR extend operations and protects the contents of its registers (including keys) from access by outside entities except via the well-defined mechanisms.
+
+- **The need to have a trusted binding between the TPM's (RTR) identity and the node's (platform) identity.** Otherwise, there is no assurance that a verification outcome is the result of applying the right verification policy to measurements of the right node.
+
+In summary, when an agent sends an attestation or otherwise accesses a Keylime API, the trust that is placed in the agent's request needs to derive from some _root identity_ (e.g., EK), bound to a particular _logical node identifier_, which the user has specified as trusted (the user must trust the identity itself as well as the binding). The node may have certain _subordinate identities_ (e.g., AK) which are transitively trusted by their binding to one of the node's root identities. This is illustrated by the below figure:
+
+<p align="center">
+    <img src="https://raw.githubusercontent.com/stringlytyped/keylime-push-proposal/379ce91b8b1f1efab9b46948bb4d04bf300afc2f/diagrams/trust-waterfall.png" alt="Diagram showing how trust in a verification result is derived from trust in a subordinate identity which in turn is derived from trust in a root identity and its binding to a logical identifier" width="800">
+</p>
+
+If the outcome of all the trust decisions shown on the diagram is favourable, we can trust that the right policy (associated with a node's logical identifier) is applied to the right attestation (produced by a subordinate cryptographic identity) by following both chains all the way to a trusted root identity.
+
+Once again, we propose two different mechanism for making these trust decisions, which users can use independently or together at their discretion:
+
+- a standard trust store mechanism for use when certificates are available to establish trust; and/or
```

Comment:
Ah, yeah, the way I've written that is not super clear. KL does currently do some certificate checking, but it is rather limited in scope and can only be used in the specific situations contemplated by KL to date (e.g., for checking EKcerts). The proposal is to genericise this functionality so it can be used for any certificate verification task, including DevIDs and any future certificate-based identity that KL may wish to support. More detail is given in [Design Details: Certificate Trust Store Mechanism](https://github.com/keylime/enhancements/blob/b1791b82100de010ce3b07f5e7c26d2bb44ae0fe/103_agent-driven-attestation.md#certificate-trust-store-mechanism)

I'll clarify this line in the proposal once I've answered your other queries.